feat: ENG-6259 - RN 0.82 Support

UncleClapton opened the pull request:
## Describe your changes

Support RN 0.82 in FSCode™

## Issue ticket number and link

https://brandingbrand.atlassian.net/browse/ENG-6259

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

## Checklist before requesting a review

- [x] A self-review of my code has been completed
- [ ] Tests have been added / updated if required
- [x] Documentation has been updated to reflect these changes

UncleClapton committed:
feat: RN 82 support

NickBurkhartBB commented on apps/example/package.json:
```diff
@@ -8,17 +8,17 @@
-    "react-native": "0.78.3",
-    "react-native-navigation": "8.4.3",
-    "react-native-safe-area-context": "5.5.2"
+    "@react-native/new-app-screen": "^0.81.5",
```
this should probably have gone to 0.82, but since we don't use it I think we can just remove it.

NickBurkhartBB commented on apps/example/package.json:
```diff
@@ -8,17 +8,17 @@
   },
   "dependencies": {
     "@brandingbrand/code-app-env": "*",
     "@brandingbrand/react-native-app-restart": "^0.4.0",
```
getting android compilation errors due to this library. may need patch or wait for a version update.

NickBurkhartBB reviewed (commented):
feel free to rebase this PR with develop

UncleClapton committed:
chore: bump yarn version

UncleClapton committed:
chore: clean up dependencies to prevent duplicate or mismatching resolutions

- React-Native dependencies used in the example app & packages are no longer installed within packages, and instead are explicitly installed in the monorepo root.
- Implicit optional peer dependencies of the app-env package are now properly listed, along with their "optional" status within "peerDependenciesMeta"
- jest is now a peer dependency of jest-config, aligning with
- As a matter of consistency. monorepo devDeps are always "*" versioned, as there is never a scenario where these versions will be looked to outside of the monorepo

UncleClapton committed:
refactor: remove excess props

UncleClapton committed:
feat(plugin-verify-dependencies): add option to install specific version when currently installed version is out of range

Also add entires for new react-native-app-restart update

UncleClapton committed:
refactor(plugin-verify-dependencies): move app-restart dep bump from 82 to 81 profile

UncleClapton committed:
chore: fix husky dep version

UncleClapton committed:
chore(example): add eslint deps as explicit dependencies

NickBurkhartBB commented on packages/plugin-verify-dependencies/src/profile/0.74.ts:
```diff
@@ -43,4 +43,13 @@ export default {
   },
-} satisfies Record<string, DependencyProfile>;
+  /**
+   * React Native Metro bundler configuration
```
doc doesn't match package

NickBurkhartBB commented on packages/plugin-verify-dependencies/src/profile/0.81.ts:
```diff
@@ -96,4 +96,11 @@ export default {
   },
-} satisfies Record<string, DependencyProfile>;
+  /**
+   * React Native Metro bundler configuration
```
doc doesn't match package

NickBurkhartBB reviewed (changes requested):
@UncleClapton tested and works, just some comment updates needed

UncleClapton committed:
fix: add updated typing to rn version script

UncleClapton committed:
docs: update documentation dependency profiles

UncleClapton committed:
docs(verify-dependencies): update jsdoc blocks for app restart profile entries
